### PR TITLE
Improve debugging

### PR DESCRIPTION
**Pull request by r0ddy**
# Description

Slow build times for developers when running app. Now, decoupled into two different build commands.
- `mvn package appengine:run` will only run the backend.
- `npm run start` will only run frontend (in dev mode).

When developing and only making frontend changes you will only have to rebuild frontend and vice versa for backend.
When depploying, run `npm run build` and then `mvn package appengine:run`.

## Type of change

- [x] Dev tool change

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by r0ddy** [comment]
> On line 72 and 73 why is column_data changed to columnData? That's is causing error. Other than that everything is cool.

I was planning to merge this after filter bug so can you review that first please?

File: frontend/src/app/components/crm/leads/leads.component.ts [hunk 1]
```diff
@@ -90,42 +90,32 @@ export class LeadsComponent implements AfterViewInit {
...
       this.dataSource.filterPredicate = (data, filter: string)  => {
-        const accumulator = (currentTerm, key) => {
-          //Using the recusrsive method here becuase using an if/else condition will increase space complexity n times.
-          return this.nestedPropertyFilterCheck(currentTerm, data, key);
+        let cleanString = (str: string): string => str.trim().toLowerCase();
+        let hasFilter = (data: object | number | string | boolean, filter: string): boolean => {
+          // traverse through JSON's tree like structure
+          if(typeof data === 'object') {
+            let keys = Object.keys(data);
+            for(let key in keys) {
```
Comment (anttsou-google) [inline frontend/src/app/components/crm/leads/leads.component.ts:98]: can this be simplified to just for (const key of Object.keys(data))?

**Comment by ataddesse** [comment]
> > On line 72 and 73 why is column_data changed to columnData? That's is causing error. Other than that everything is cool.
> 
> I was planning to merge this after filter bug so can you review that first please?

Sure, my bad.

**Comment by r0ddy** [comment]
> LGTM, will the dev web app still be accessed the same (both front end and back end at localhost:8080) or does this change that?

Yeah so the dev web app will now load on localhost:4200. So that you can hot-reload frontend without rebuilding backend. You can still run `npm run builddev` which will populate the webapp folder, but this is now optional for backend developers and no longer runs as part of the Maven build.

File: frontend/src/app/components/crm/leads/leads.component.ts (hunk 1 above)
Comment (r0ddy) [reply frontend/src/app/components/crm/leads/leads.component.ts]: Yup, has been done!

**Pull request merged**